lib/teleterm/apiserver/grpccredentials.go

```diff
@@ -0,0 +1,93 @@
+// Copyright 2022 Gravitational, Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package apiserver
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"os"
+	"path/filepath"
+
+	"github.com/gravitational/teleport/api/utils/keys"
+	"github.com/gravitational/teleport/lib/utils"
+
+	"github.com/gravitational/trace"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials"
+)
+
+const (
+	// tshdCertFileName is the file name of the cert created by the tshd process. The Electron app
+	// expects it to exist under this name in the certs dir passed through a flag to tshd.
+	tshdCertFileName = "tshd.crt"
+	// rendererCertFileName is the file name of the cert created by the renderer process of the
+	// Electron app.
+	rendererCertFileName = "renderer.crt"
+)
+
+// createServerCredentials creates mTLS credentials for a gRPC server. The client cert file is read
+// only on an incoming connection, not upfront.
```

Comment:
I can see that the client cert file is read on every connection, the comment should probably explain _why_

Reply:
> I can see that the client cert file is read on every connection, the comment should probably explain _why_

Comment added:

https://github.com/gravitational/teleport/blob/fddbb94a3da5f0318c125c9833a05a264888e708/lib/teleterm/apiserver/grpccredentials.go#L40-L43